src/dataframerow/utils.jl

```diff
@@ -149,16 +149,36 @@ function row_group_slots(cols::Tuple{CategoricalVector},
         # we could simply copy refs to groups, but the performance gain is negligible,
         # so always sort groups in the order of levels
         refmap = [0; CategoricalArrays.order(col.pool)]
+        seen = fill(false, length(refmap))
         if skipmissing
             refmap .+= 1
+            seen[1] = true
         else
             refmap[1] = ngroups
         end
         @inbounds for i in eachindex(groups)
-            groups[i] = refmap[col.refs[i]+1]
+            j = refmap[col.refs[i]+1]
+            groups[i] = j
+            seen[j] |= true
```

Comment:
why `|=` and not simply `=`? Is it faster?

Reply:
I figured it could avoid a write when not needed, but apparently it's slightly slower:
```julia
function f(refmap, groups, refs)
    seen = fill(false, length(refmap))
    @inbounds for i in eachindex(groups)
        j = refmap[refs[i]+1]
        groups[i] = j
        seen[j] |= true
    end
    seen, groups
end

N = 100_000
ngroups = 10
groups = Vector{Int}(undef, N)
refs = rand(1:ngroups, N)
refmap = rand(1:ngroups+1, ngroups+1)

# With |=
julia> @btime f(refmap, groups, refs);
  163.944 μs (2 allocations: 128 bytes)

# With =
julia> @btime f(refmap, groups, refs);
  149.227 μs (2 allocations: 128 bytes)
```